pkg/configuration/parameters_test.go

```diff
@@ -1,317 +1,104 @@
+	tParam = TestJiraAPIResourceParameters{}
 
-	testExpectedBoolResult(t, param.Meta.MultipleIssue, false)
-}
+	err := faker.FakeData(&tParam)
+	require.NoError(t, err, "problem creating fake data")
 
-func TestInitializeIssueListMultiple(t *testing.T) {
-	param := &JiraAPIResourceParameters{}
+	tParam.Destination = tParam.Destination + "/"
+	context = "ReadIssue"
+	issueList = "ABC-123 DEF-456"
 
-	list := "ABC-001,DEF-9999"
-	param.initializeIssueList(&list)
+	fPtr = new(bool)
+	*fPtr = false
+	tPtr = new(bool)
+	*tPtr = true
 
-	if param.IssueList == nil {
-		t.Errorf("IssueList was nil but should have contained at least one element")
+	return func(t *testing.T) {
+		t.Log("teardown test cases...")
 	}
-
-	testExpectedBoolResult(t, param.Meta.MultipleIssue, true)
 }
 
-func TestInitializeContext(t *testing.T) {
-	dummyString := "abc"
-	readIssueString := "ReadIssue"
-	editCustomFieldString := "EditCustomField"
+func TestPostParse(t *testing.T) {
+	teardown := setup(t)
+	defer teardown(t)
 
-	tables := []struct {
-		i *string
-		o Context
-	}{
-		{nil, Unknown},
-		{&dummyString, Unknown},
-		{&readIssueString, ReadIssue},
-		{&editCustomFieldString, EditCustomField},
-	}
+	t.Run("app parameters VALID and READY from VALID input flags and param", func(t *testing.T) {
+		t.Skip()
 
-	for _, table := range tables {
-		param := &JiraAPIResourceParameters{}
-		param.initializeContext(table.i)
+		// Arrange
+		param := configuration.JiraAPIResourceParameters{}
+		param.Parse()
+		param = convertToJiraApiResourceParameters()
 
-		if param.Context != table.o {
-			t.Errorf("Context value was incorrect, got: %v, want: %v.", param.Context, table.o)
-		}
-	}
-}
+		// Act
+		param.InitializeAndValidatePostParse(&context, &issueList)
 
-func TestParsePtrParameters(t *testing.T) {
-	tables := []struct {
-		in1  string
-		in2  string
-		out1 string
-	}{
-		{"url", "github.com", "JiraAPIUrl"},
-		{"username", "I_AM_USER", "Username"},
-		{"password", "SECRET_PWD", "Password"},
-		{"customFieldName", "Custom_Field_Name", "CustomFieldName"},
-		{"customFieldType", "String", "CustomFieldType"},
-		{"customFieldValue", "123abc", "CustomFieldValue"},
-		{"customFieldValueFromFile", "filename.dummy", "CustomFieldValueFromFile"},
-		{"loggingLevel", "TEST", "LoggingLevel"},
-	}
+		// Assert
+		assert.True(t, param.Meta.AllMandatoryValuesPresent(), "method AllMandatoryValuesPresent() returned false")
+		assert.True(t, param.Meta.Ready(), "method Ready() returned false")
+	})
 
-	for _, table := range tables {
-		param := &JiraAPIResourceParameters{}
-		runFlagTest(t, param, table.in1, table.in2, table.out1)
-	}
 }
 
-func runFlagTest(t *testing.T, param *JiraAPIResourceParameters, p, expected string, refVal string) {
-	os.Args = setupOsArgs(fmt.Sprintf("--%s=%s", p, expected))
-
-	param.Parse()
-
-	rParam := reflect.Indirect(reflect.ValueOf(param))
-	f := rParam.FieldByName(refVal)
+func convertToJiraApiResourceParameters() configuration.JiraAPIResourceParameters {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [func convertToJiraApiResourceParameters should be convertToJiraAPIResourceParameters](https://app.codacy.com/manual/TurnsCoffeeIntoScripts/jira-api-resource/pullRequest?prid=5035521)